src/components/logs/logsWebview.ts

```diff
@@ -0,0 +1,145 @@
+import * as path from 'path';
+import * as vscode from 'vscode';
+
+export class LogsPanel {
+    public static readonly viewType = 'vscodeKubernetesLogs';
+    public static currentPanels = [];
+
+    private disposables: vscode.Disposable[] = [];
+    private content: string;
+    private resource: string;
+
+    public static createOrShow(extensionPath: string, content: string, resource: string): LogsPanel {
+        const column = vscode.window.activeTextEditor ? vscode.window.activeTextEditor.viewColumn : undefined;
+
+        // If we already have a panel, show it.
+        const currentPanel = LogsPanel.currentPanels[resource];
+        if (currentPanel) {
+            currentPanel.setInfo(content, resource);
+            currentPanel.update();
+            currentPanel.panel.reveal(column);
+            return currentPanel;
+        }
+        const panel = vscode.window.createWebviewPanel(LogsPanel.viewType, "Kubernetes Logs", column || vscode.ViewColumn.One, {
+            enableScripts: true,
+
+            // And restrict the webview to only loading content from our extension's `media` directory.
+            localResourceRoots: [
+            ],
+        });
+
+        const result = new LogsPanel(panel, content, resource);
+        LogsPanel.currentPanels[resource] = result;
+        return result;
+    }
+
+    private constructor(
+        private readonly panel: vscode.WebviewPanel,
+        content: string,
+        resource: string
+    ) {
+        this.content = content;
+        this.resource = resource;
+
+        this.update();
+        this.panel.onDidDispose(() => this.dispose(), null, this.disposables);
+
+        this.panel.onDidChangeViewState((e: any) => {
+            if (this.panel.visible) {
+                this.update();
+            }
+        }, null, this.disposables);
+    }
+
+    public setInfo(content: string, resource: string) {
+        this.content = content;
+        this.resource = resource;
+        this.update();
+    }
+
+    public dispose() {
+        delete LogsPanel.currentPanels[this.resource];
+
+        this.panel.dispose();
+
+        while (this.disposables.length) {
+            const x = this.disposables.pop();
+            if (x) {
+                x.dispose();
+            }
+        }
+    }
+
+    private update() {
+        this.panel.title = `Kubernetes logs ${this.resource}`;
```

Comment:
Logs with a capital L?  Perhaps also a separator?  To make it easy to scan I'd even suggest `Logs - ${this.resource}` so that less of it gets elided away when there are lots of tabs.

Reply:
done.